webpack/assets/javascripts/react_app/components/common/table/index.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import ReactTable from 'react-table';
+import 'react-table/react-table.css';
+import helpers from '../../../common/helpers';
+
+const SETTING_PER_PAGE = 10; // TODO: use foreman per_page settings
+const BLANK_ROWS = 3;
+const TOOLTIP_SIZE = 15;
+
+class Table extends React.Component {
+  isEmpty() {
+    return this.props.data.length === 0;
+  }
+
+  render() {
+    const { columns, data, showPagination } = this.props;
+    const rows = this.isEmpty() ? BLANK_ROWS : data.length;
+
+    return (
+      <ReactTable
+        className="-striped"
+        defaultPageSize={
+          showPagination
+            ? Number(helpers.getQueryParam('per_page')) || Math.min(rows, SETTING_PER_PAGE)
+            : rows
+        }
+        showPagination={showPagination}
+        data={data}
+        columns={columns}
+        // i18n
+        previousText={__('Previous')}
+        nextText={__('Next')}
+        pageText={__('page')}
+        ofText={__('of')}
+        rowsText={__('rows')}
+        noDataText={__('No data found')}
```

Comment:
should probably use empty state optionally (in case the table is the main part of the view).